Maple2.Server.Game/Manager/ClubManager.cs

```diff
@@ -0,0 +1,250 @@
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Error;
+using Maple2.Model.Game;
+using Maple2.Model.Game.Club;
+using Maple2.Server.Core.Sync;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+using Maple2.Server.Game.Util.Sync;
+using Maple2.Tools.Extensions;
+using Serilog;
+
+namespace Maple2.Server.Game.Manager;
+
+public class ClubManager : IDisposable {
+    private readonly GameSession session;
+
+    public Club? Club { get; private set; }
```

Comment:
I feel like we should find a way to avoid making this nullable. This is different from guilds because the `GuildManager` still exists when you don't have a guild, while the `ClubManager` would not exist if you don't have a club? (Since there's 1 `ClubManager` per club)

Reply:
Ideas:
1. Make Club `init` only
2. Make `SetClub` private (don't seem to be called anywhere else
3. Mark `SetClub` as `[SetsRequiredMembers]`

I think this would be enough to remove nullable

Reply:
I think `SetRequiredMembers` looks like it's only for constructors.

```Attribute 'System.Diagnostics.CodeAnalysis.SetsRequiredMembersAttribute' is not valid on this declaration type. It is valid on 'Constructor' declarations only.```

I could just move all of `SetClub` within the constructor?

Reply:
Yea it still doesn't work because the constructor doesn't allow failures, i think what you want is a static function on `ClubManager` called `public static ClubManager? Create(...)` and make the constructor private.